astropy/io/ascii/cds.py

```diff
@@ -164,14 +203,493 @@ def get_cols(self, lines):
+        # to the hyphen using a format string.
+        byte_count_width = len(str(sum(widths) + len(self.cols) - 1))
+
+        # Format string for Start Byte and End Byte
+        singlebfmt = "{:" + str(byte_count_width) + "d}"
+        fmtb = singlebfmt + "-" + singlebfmt
+        # Add trailing single whitespaces to Bytes column for better visibility.
+        singlebfmt += " "
+        fmtb += " "
+
+        # Set default width of Label and Description Byte-By-Byte columns.
+        max_label_width, max_descrip_size = 7, 16
+
+        bbb = Table(names=['Bytes', 'Format', 'Units', 'Label', 'Explanations'],
+                    dtype=[str]*5)
+
+        # Iterate over the columns to write Byte-By-Byte rows.
+        for i, col in enumerate(self.cols):
+            # Check if column is MaskedColumn
+            col.has_null = isinstance(col, MaskedColumn)
+
+            # Check if the column format was set by the passed ``formats`` argument.
+            if col.format is None:
+                if col.info.format is not None:
+                    col.format = col.info.format
+
+            if col.format is not None:
+                col.meta.size = max([len(sval) for sval in col.str_vals])
+
+            # Set CDSColumn type, size and format.
+            if np.issubdtype(col.dtype, np.dtype(int).type) or np.issubdtype(col.dtype, np.integer):
+                # Integer formatter
+                self._set_column_val_limits(col)
+                if getattr(col.meta, 'size', None) is None:  # If ``formats`` not passed.
+                    col.meta.size = max(len(str(col.max)), len(str(col.min)))
+                col.fortran_format = "I" + str(col.meta.size)
+                col.format = ">" + col.fortran_format[1:]
+
+            elif np.issubdtype(col.dtype, np.dtype(float).type):
+                # Float formatter
+                self._set_column_val_limits(col)
+                self.column_float_formatter(col)
+
+            else:
+                # String formatter, ``np.issubdtype(col.dtype, str)`` is ``True``.
+                dtype = col.dtype.str
+                if col.has_null:
+                    mcol = col
+                    mcol.fill_value = ""
+                    coltmp = Column(mcol.filled(), dtype=str)
+                    dtype = coltmp.dtype.str
+                if getattr(col.meta, 'size', None) is None:  # If ``formats`` not passed.
+                    col.meta.size = int(re.sub(r'^[^0-9]+(\d+)$', r'\1', dtype))
+                col.fortran_format = "A" + str(col.meta.size)
+                col.format = str(col.meta.size) + "s"
+
+            endb = col.meta.size + startb - 1
+
+            # ``mixin`` columns converted to string valued columns will not have a name
+            # attribute. In those cases, a ``Unknown`` column label is put, indicating that
+            # such columns can be better formatted with some manipulation before calling
+            # the CDS/MRT writer.
+            if col.name is None:
```

Comment:
Points like this are exactly why we asked you, @taldcroft, to have a look! I'm not using (or writing code for) QTables.

Reply:
When I check `col.name` and `col.info.name` for the `qt` above, both of them are set to `None`. Whereas, `qt.info()` gives out `a float64    m Quantity`.

This may possibly be because the [`Basereader.write` func in `core.py` ](https://github.com/astropy/astropy/blob/5942ac3a222045c26db0ab4497d6128d1a9dc80c/astropy/io/ascii/core.py#L1447) takes the table columns and saves them as a list in `header.cols` using `table.columns.values()`.

For a `Table`, `t.columns.values()` is `odict_values([<Column name='b' dtype='float64' length=2>
2.123456789
   2.45e+20])`
For a `QTable`, `qt.columns.values()` is `odict_values([<Quantity [1., 2.] m>])`

Reply:
We had not foreseen that QTables may not work with the code. Thank you for your valuable comment. :)
Looks like this may need some work. So, **needs an issue**.